File: src/main/java/my/study/repository/design/pattern/factory/UseDemo.java
package my.study.repository.design.pattern.factory;

/**
 * @author djh on  2018/7/28 23:09
 * @E-Mail 1544579459@qq.com
 */
public class UseDemo {
    public static void main(String[] args) {
        BMW bmwThree = EasyFactory.getBMWThree();
        BMW bmwSix = EasyFactory.getBMWSix();

        bmwThree.speak();
        bmwSix.speak();
    }
}

File: src/main/java/my/study/repository/design/pattern/factory/kotlin/UseDemo.kt
package my.study.repository.design.pattern.factory.kotlin

/**
 * @author djh on  2018/10/5 18:53
 * @E-Mail 1544579459@qq.com
 */
object UseDemo {
    @JvmStatic
    fun main(args: Array<String>) {
        val bmwThree = EasyFactory.getBmwThree()
        val bmwSix = EasyFactory.getBmwSix()
        
        bmwThree.speak()
        bmwSix.speak()
        
    }
}